### PR TITLE
Noop refactor to make MSVC compiler not crash with internal error

### PR DESCRIPTION
**Pull request by p2004a**
MSVC 17.4.5 crashes with internal compilation error when trying to build RelWithDebInfo:

```
25>C:\Users\Marek\Workspace\spring\rts\Game\SelectedUnitsAI.cpp(590): fatal error C1001: Internal compiler error.
25>(compiler file 'D:\a\_work\1\s\src\vctools\Compiler\Utc\src\p2\main.c', line 224)
25> To work around this problem, try simplifying or changing the program near the locations listed above.
25>If possible please provide a repro here: https://developercommunity.visualstudio.com
25>Please choose the Technical Support command on the Visual C++
25> Help menu, or open the Technical Support help file for more information
25>  link!RaiseException()+0x6c
25>  link!RaiseException()+0x6c
25>  link!InvokeCompilerPassW()+0x87d1b
25>  link!InvokeCompilerPassW()+0xf86f1
25>
25>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(1096,5): error MSB6006: "link.exe" exited with code -529706956.
25>Done building project "engine-legacy.vcxproj" -- FAILED.
```

so I've rewritten the code as little as possible to get rid of this issue...

### CODE REVIEW

**Comment by lhog** [comment]
Hmm very interesting and trivial place to stumble upon. Let me updated to the latest MSVC and see how it goes.

**Comment by lhog** [comment]
Hmm doesn't reproduce for me without the proposed changes:
image

**Comment by p2004a** [comment]
I will write reproduction steps and try to reproduce on some other machine on my side.

**Comment by p2004a** [comment]
I've reproduced it on my second machine, steps to repro inside of empty build directory, on HEAD, with up to date vclibs64:

```
cmake -DAI_TYPES=NATIVE -DBUILD_spring-dedicated=OFF -DBUILD_spring-headless=OFF -DENABLE_DOCS=OFF -G "Visual Studio 17 2022" .. > configure.txt
cmake --build . --config RelWithDebInfo --target engine-legacy > build.txt
```

and those outs:
- [build.txt](https://github.com/beyond-all-reason/spring/files/10771614/build.txt)
- [configure.txt](https://github.com/beyond-all-reason/spring/files/10771617/configure.txt)

If you are configuring somehow differently, please share so I can reproduce with the same options.

File: rts/Game/SelectedUnitsAI.cpp [hunk 1]
```diff
@@ -536,12 +536,14 @@ bool CSelectedUnitsHandlerAI::SelectAttackNet(const Command& cmd, int playerNum)
...
+	unsigned int realCount = 0;
 
-	if (selectedCount == 0)
-		return ret;
+	for (int unitID : netSelectedUnitIDs)
+		if (unitHandler.GetUnit(unitID) != nullptr)
+			++realCount;
 
+	if (realCount == 0)
+		return ret;
```
[the remark this answers is not in the document]
Comment (p2004a) [reply rts/Game/SelectedUnitsAI.cpp:545]: not sure how you suggest to use `std::accumulate`, I've pushed change that uses `std::count_if` if that is preferred patter.

[the remark this answers is not in the document]
Comment (lhog) [reply rts/Game/SelectedUnitsAI.cpp:545]: Sure `std::count_if` was appropriate

**Pull request merged**